### ISSUE TITLE
ENH: Make Polynomial adhere to set_printoptions 

### ISSUE THREAD

**Issue by axil** [numpy/numpy#21653]
### Proposed new feature or change:

Currently Polynomial displays as many digits as it can:

```
x = np.arange(10)
y = 5 * x + 1 + np.random.randn(10)
f = np.polynomial.Polynomial.fit(x, y, 1).convert(); f
```
𝑥↦0.622720266227418+5.0153841318332235𝑥

It'd be better if it would not display more digits than configured with `np.set_printoptions`:

𝑥↦0.62272026+5.01538413𝑥

(or less if it can be uniquely represented).

**Comment by eendebakpt** [comment]
@axil I like the usage of `np.printoptions` for the formatting.

In https://github.com/axil/numpy/compare/polynomial-printoptions...eendebakpt:polynomial_str_printoptions I use your branch to make the presentation of the polynomial show the domain mapping.
```
import numpy as np
from numpy import printoptions
import numpy.polynomial.polyutils as pu

p =np.polynomial.Polynomial([4,1/3,3], symbol='z', domain=[0,2.0000001])
print(p)
with printoptions(precision=3):
    print(str(p))
```
becomes
```
<module 'numpy' from '/mnt/data/numpy/numpy/__init__.py'>
4.0 + 0.33333333·(-1.0 + 0.99999995z)¹ + 3.0·(-1.0 + 0.99999995z)²
4.0 + 0.333·(-1.0 + 1.0z)¹ + 3.0·(-1.0 + 1.0z)²
```
Once your PR is merged I will push mine. (merging would make the PR a bit too complex I think)

**Comment by axil** [comment]
Made switching to exponential notation more 'natural':

![image](https://user-images.githubusercontent.com/170910/172213861-ec2167a3-304a-4b5c-b342-363def1d424b.png)

**Comment by axil** [comment]
Here's the complete list of `printoptions` annotated in relation to polynomials:

• precision: implemented
• threshold: not applicable
• edgeitems: not applicable
• linewidth: the only `printoption` that was already supported before this PR, I didn't touch it.
• suppress: not applicable
• nanstr: implemented
• infstr: implemented
• sign: not applicable
• formatter: not applicable
• floatmode: only 'fixed' (=always `precision` digits) and anything else (=enough to be uniquely represented, max `precision`) with `maxprec_equal` as default (which counts as 'anything else')
• legacy: not applicable

**Comment by HaoZeke** [comment]
> I've raised the threshold of switching to exponential notation (for polynomials only) to 10^9 and that made the tests pass.

Might be better to fix the tests to use the 10^8 notation instead of increasing the threshold.

**Comment by axil** [comment]
> > I've raised the threshold of switching to exponential notation (for polynomials only) to 10^9 and that made the tests pass.
> 
> Might be better to fix the tests to use the 10^8 notation instead of increasing the threshold.

I've had a look at the failing test and there was nothing peculiar about them using 123456789 as a constant. The 12345678 can work equally well there. Fixed the tests and switched to 10^8 threshold.

**Comment by axil** [comment]
Some tests are failing now. Need some time to look into it.

**Comment by axil** [comment]
Merged PR #21696 that fixes #21695 (replace x¹ with x). All tests are passing (except Travis that looks frozen). Ready to merge.

**Comment by mattip** [comment]
Something is off here: I see comments on this issue that seem to refer to a PR, not an issue. Comments about reviewing PR code changes, needed tests, or work done in a PR belong on the PR. Comments describing and discussing the problem driving the change belong in the issue, but I think we all agreed using printoptions for polynomial `repr` is desired.

**Comment by axil** [comment]
@mattip I agree that maybe some of the comments could be posted elsewhere. But I don't quite understand what do you propose to do now. Polynomial `repr` uses printoptions in the master branch already. This PR does not deal with it. In addition to `repr`, polynomials have the following three representation modes:

![image](https://user-images.githubusercontent.com/170910/173239350-b8c3bad5-30a6-4fa0-8445-02221b4b7653.png)

The PR deals with:
– unicode mode of str(p) (default on Linux) 
– ascii mode of str(p) (default on Windows)
– latex mode of display(p) (default in jupyter notebooks)

Anything wrong with it?

**Comment by mattip** [comment]
I was reacting to the "ready to merge" comment above. Tests are passing on #21654 and it is ready for review.
